ee/server/service/users.go

```diff
@@ -0,0 +1,38 @@
+package service
+
+import (
+	"context"
+	"errors"
+
+	"github.com/fleetdm/fleet/v4/server/contexts/ctxerr"
+	"github.com/fleetdm/fleet/v4/server/fleet"
+	"github.com/fleetdm/fleet/v4/server/ptr"
+)
+
+// GetSSOUser is the premium implementation of svc.GetSSOUser, it allows to
+// create users during the SSO flow the first time they log in if
+// config.SSOSettings.EnableJITProvisioning is `true`
+func (svc *Service) GetSSOUser(ctx context.Context, auth fleet.Auth) (*fleet.User, error) {
+	config, err := svc.ds.AppConfig(ctx)
+	if err != nil {
+		return nil, ctxerr.Wrap(ctx, err, "getting app config")
+	}
+
+	user, err := svc.Service.GetSSOUser(ctx, auth)
+	var nfe fleet.NotFoundError
+	if !errors.As(err, &nfe) || !config.SSOSettings.EnableJITProvisioning {
+		return user, err
+	}
+
+	user, err = svc.Service.NewUser(ctx, fleet.UserPayload{
+		Name:       ptr.String(auth.UserDisplayName()),
+		Email:      ptr.String(auth.UserID()),
+		SSOEnabled: ptr.Bool(true),
+		GlobalRole: ptr.String(fleet.RoleObserver),
+	})
+	if err != nil {
+		return nil, ctxerr.Wrap(ctx, err, "creating new SSO user")
+	}
```

Comment:
Should we add an activity here? (I believe there's a push to add more activity logging moving forward, related issue: #6610).

Reply:
Great idea!

```
@lucasmrod was added to Fleet by SSO
```
perhaps?

Reply:
w00t! is this something we need to discuss with product or can I go ahead and update this PR?

Reply:
I'd say go for it!

Reply:
gotcha, back-end changes added in [900c22a](https://github.com/fleetdm/fleet/pull/7182/commits/900c22a6e2d4ad06a99d06a896765b627f97eeda), UI changes will come along in the front-end PR.